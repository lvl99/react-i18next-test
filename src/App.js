import React, { useState, useCallback, useEffect } from "react";
import { useTranslation } from "react-i18next";
import logo from "./logo.svg";
import "./App.css";

export default function App({ language }) {
  const { t, i18n } = useTranslation();
  const [lng, setLng] = useState(language);

  const handleLanguageChange = useCallback(
    newLanguage => {
      if (newLanguage !== lng) {
        i18n.changeLanguage(newLanguage);
        setLng(newLanguage);
      }
    },
    [lng]
  );

  useEffect(() => {
    if (lng !== i18n.language) {
      setLng(i18n.language);
    }
  }, [lng, i18n.language]);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1>{t("welcome.toReact", { name: "Test" })}</h1>
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <button onClick={() => handleLanguageChange("en")}>Language: EN</button>
        <button onClick={() => handleLanguageChange("fr")}>Language: FR</button>
      </header>
    </div>
  );
}
