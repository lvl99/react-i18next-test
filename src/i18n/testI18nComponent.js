import React from "react";

const mockIntl = {
  changeLanguage: () => {}
};

const mockTranslate = key => key;
const mockTranslateWithValues = (key, values) =>
  `${key}:${JSON.stringify(values)}`;

export const testI18nComponent = Component => {
  const TestI18nComponent = props => {
    const tProps = {
      ...props,
      intl: mockIntl,
      t: mockTranslate
    };
    return <Component {...tProps} />;
  };

  return TestI18nComponent;
};

export const testI18nComponentValues = Component => {
  const TestI18nComponent = props => {
    const tProps = {
      ...props,
      intl: mockIntl,
      t: mockTranslateWithValues
    };
    return <Component {...tProps} />;
  };

  return TestI18nComponent;
};

export default testI18nComponent;
