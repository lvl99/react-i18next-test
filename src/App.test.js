import React from "react";
import ReactDOM from "react-dom";
import { mount, shallow } from "enzyme";
import App from "./App";
import testI18nComponent from "./i18n/testI18nComponent";

const Wrapper = testI18nComponent(App);

it("matches shallow snapshot", () => {
  const wrapper = shallow(<Wrapper />);
  expect(wrapper).toMatchSnapshot();
});

it("matches mount snapshot", () => {
  const wrapper = mount(<Wrapper />);
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  expect(() => {
    const div = document.createElement("div");
    ReactDOM.render(<Wrapper />, div);
    ReactDOM.unmountComponentAtNode(div);
  }).not.toThrow();
});
