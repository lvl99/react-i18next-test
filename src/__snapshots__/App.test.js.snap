// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches mount snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <TestI18nComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "intl": Object {
          "changeLanguage": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="App"
        >
          <header
            class="App-header"
          >
            <img
              alt="logo"
              class="App-logo"
              src="logo.svg"
            />
            <h1>
              welcome.toReact
            </h1>
            <button>
              Language: EN
            </button>
            <button>
              Language: FR
            </button>
          </header>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h1>
              welcome.toReact
            </h1>
            <button
              onClick={[Function]}
            >
              Language: EN
            </button>
            <button
              onClick={[Function]}
            >
              Language: FR
            </button>
          </header>,
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <header
              class="App-header"
            >
              <img
                alt="logo"
                class="App-logo"
                src="logo.svg"
              />
              <h1>
                welcome.toReact
              </h1>
              <button>
                Language: EN
              </button>
              <button>
                Language: FR
              </button>
            </header>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg"
                />,
                <h1>
                  welcome.toReact
                </h1>,
                <button
                  onClick={[Function]}
                >
                  Language: EN
                </button>,
                <button
                  onClick={[Function]}
                >
                  Language: FR
                </button>,
              ],
              "className": "App-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <img
                  alt="logo"
                  class="App-logo"
                  src="logo.svg"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "App-logo",
                  "src": "logo.svg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
              Object {
                "instance": <h1>
                  welcome.toReact
                </h1>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "welcome.toReact",
                },
                "ref": null,
                "rendered": Array [
                  "welcome.toReact",
                ],
                "type": "h1",
              },
              Object {
                "instance": <button>
                  Language: EN
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Language: EN",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Language: EN",
                ],
                "type": "button",
              },
              Object {
                "instance": <button>
                  Language: FR
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Language: FR",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Language: FR",
                ],
                "type": "button",
              },
            ],
            "type": "header",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "intl": Object {
            "changeLanguage": [Function],
          },
          "t": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="App"
          >
            <header
              class="App-header"
            >
              <img
                alt="logo"
                class="App-logo"
                src="logo.svg"
              />
              <h1>
                welcome.toReact
              </h1>
              <button>
                Language: EN
              </button>
              <button>
                Language: FR
              </button>
            </header>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <header
              className="App-header"
            >
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />
              <h1>
                welcome.toReact
              </h1>
              <button
                onClick={[Function]}
              >
                Language: EN
              </button>
              <button
                onClick={[Function]}
              >
                Language: FR
              </button>
            </header>,
            "className": "App",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <header
                class="App-header"
              >
                <img
                  alt="logo"
                  class="App-logo"
                  src="logo.svg"
                />
                <h1>
                  welcome.toReact
                </h1>
                <button>
                  Language: EN
                </button>
                <button>
                  Language: FR
                </button>
              </header>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
                  />,
                  <h1>
                    welcome.toReact
                  </h1>,
                  <button
                    onClick={[Function]}
                  >
                    Language: EN
                  </button>,
                  <button
                    onClick={[Function]}
                  >
                    Language: FR
                  </button>,
                ],
                "className": "App-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <img
                    alt="logo"
                    class="App-logo"
                    src="logo.svg"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "logo",
                    "className": "App-logo",
                    "src": "logo.svg",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "img",
                },
                Object {
                  "instance": <h1>
                    welcome.toReact
                  </h1>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "welcome.toReact",
                  },
                  "ref": null,
                  "rendered": Array [
                    "welcome.toReact",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": <button>
                    Language: EN
                  </button>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Language: EN",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Language: EN",
                  ],
                  "type": "button",
                },
                Object {
                  "instance": <button>
                    Language: FR
                  </button>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Language: FR",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Language: FR",
                  ],
                  "type": "button",
                },
              ],
              "type": "header",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`matches shallow snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TestI18nComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "intl": Object {
        "changeLanguage": [Function],
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "intl": Object {
          "changeLanguage": [Function],
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
